contrib/passwd.bzl

```diff
@@ -54,17 +39,28 @@ passwd_file = rule(
         "info": attr.string(default = "user"),
         "home": attr.string(default = "/home"),
         "shell": attr.string(default = "/bin/bash"),
-        "build_tar": attr.label(
-            default = Label("@bazel_tools//tools/build_defs/pkg:build_tar"),
-            cfg = "host",
-            executable = True,
-            allow_files = True,
-        ),
     },
     executable = False,
     outputs = {
-        "out": "%{name}.passwd",
-        "tar": "%{name}.passwd.tar",
+        "out": "%{name}",
     },
     implementation = _impl,
 )
+
+def passwd_tar(name, username, uid, gid, info, home, shell, passwd_file_name=""):
+    tar_name = "%s" %name
+    file_name = passwd_file_name or "%s.file" %name
+    passwd_file(
+        name = file_name,
+        username = username,
+        uid = uid,
+        gid = gid,
+        info = info,
+        home = home,
+        shell = shell,
+    )
+    pkg_tar(
```

Comment:
I don't think there is a way to make this rule emit a tarball with `/etc/passwd` since `passwd_file`s name can't have a `/` in it...

Reply:
the pkg_tar package_dir attribute makes it possible.
You can check my tests to confirm.